### PR TITLE
runtime(nohlsearch): add g:loaded_nohlsearch guard

### PR DESCRIPTION
**Pull request by habamax**
Relates to https://github.com/vim/vim/issues/15039#issuecomment-2260305256

### CODE REVIEW

File: runtime/pack/dist/opt/nohlsearch/plugin/nohlsearch.vim [hunk 1]
```diff
@@ -5,6 +5,12 @@
 " turn off hlsearch after:
 " - doing nothing for 'updatetime'
 " - getting into insert mode
```
Comment (zeertzjq) [inline runtime/pack/dist/opt/nohlsearch/plugin/nohlsearch.vim:7]: These comments seem to fit better just above the autocommands, not here.

Comment (zeertzjq) [reply runtime/pack/dist/opt/nohlsearch/plugin/nohlsearch.vim:7]: Actually that's not necessarily true, as they can serve as the description of the behavior of the file here.

Comment (habamax) [reply runtime/pack/dist/opt/nohlsearch/plugin/nohlsearch.vim:7]: well :)

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**